File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
public class Solution {
    public String reversePrefix(String word, char ch) {
        
        
        
        //1. Determine the index of the particular array
        int index = word.indexOf(ch);
        if(index == -1){
            return word;
        }
        
        //2. stringbuildef
        StringBuilder resultSB = new StringBuilder();
        
        for(int i = 0;i<word.length();i++){
            if(i<=index){
                resultSB.append(word.charAt(i));
                if(i == index){
                    resultSB.reverse();
                }
            }
            else{
                resultSB.append(word.charAt(i));
            }
        }
        
        //3. return the answer
        return resultSB.toString();
        
    }
}

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        #1. Convert into a list 
        wordList = list(word)
        outputWord = ""
        
        #2. determine the index where ch is there
        try:
            indexPresent = wordList.index(ch)
        except ValueError:
            return word
        
        #3. check if you can find out to reverse it
        if indexPresent!=-1:
            outputWord = ''.join(reversed(wordList[0:indexPresent+1]))
            outputWord += ''.join(wordList[indexPresent+1:])
        
        #4. return the output word
        return outputWord